crates/viewer/re_viewer/src/app.rs

```diff
@@ -1994,168 +1975,222 @@ impl App {
                 }
             };
 
-            let store_id = msg.store_id();
+            match msg {
+                DataSourceMessage::LogMsg(msg) => {
+                    self.receive_log_msg(&msg, store_hub, egui_ctx, &channel_source);
+                }
 
-            if store_hub.is_active_blueprint(store_id) {
-                // TODO(#5514): handle loading of active blueprints.
-                re_log::warn_once!(
-                    "Loading a blueprint {store_id:?} that is active. See https://github.com/rerun-io/rerun/issues/5514 for details."
-                );
+                DataSourceMessage::UiCommand(ui_command) => {
+                    self.receive_data_sourceui_command(ui_command, &channel_source);
```

Comment:
I think this will schedule the commands to be executed at the end of the frame. Maybe we should add a call to `run_pending_system_commands` directly after `receive_messages` to avoid a frame delay?